soc/arm/bcm_vk/viper/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_include_directories(.)
+zephyr_sources(
+	soc.c
+)
+
+zephyr_library_sources_ifdef(CONFIG_ARM_MMU mmu_regions.c)
```

Comment:
Not sure why there is a mix of using both `zephyr_sources` and `zephyr_library_sources`.
As there has been no `zephyr_library()` call, then the file `mmu_regions.c` will also go to `libzephyr.a`.

So I suggest to use either:
```
zephyr_sources(
	soc.c
)

zephyr_sources_ifdef(CONFIG_ARM_MMU mmu_regions.c)
```
or
```
zephyr_library()
zephyr_library_sources(
	soc.c
)

zephyr_library_sources_ifdef(CONFIG_ARM_MMU mmu_regions.c)
```
if you prefer to have those files in a dedicated lib.

Reply:
Thanks. Not sure how I got to using `zephyr_library_sources_ifdef`.. no intention to put in dedicated lib, will switch to `zephyr_sources_ifdef`.